test/stripe/charge_test.rb

```diff
@@ -95,5 +95,43 @@ class ChargeTest < Test::Unit::TestCase
       })
       assert c.paid
     end
+
+    should "properly handle an array or dictionaries" do
+      @mock.expects(:post).with do |url, api_key, params|
+        url == "#{Stripe.api_base}/v1/charges" && api_key.nil? && CGI.parse(params) == {
+          'currency' => ['usd'], 'amount' => ['100'],
+          'source' => ['btcrcv_test_receiver'],
+          'level3[][red]' => ['firstred', 'another'],
+          'level3[][one]' => ['fish'],
+        }
+      end.once.returns(test_response(test_charge))
+
+      c = Stripe::Charge.create({
+        :amount => 100,
+        :source => 'btcrcv_test_receiver',
+        :currency => "usd",
+        :level3 => [{red: 'firstred'}, {one: 'fish', red: 'another'}]
+      })
+      assert c.paid
+    end
+
+    should "TODO remove.. just showing that the encoding is ambiguous. The output is the same to a different input" do
```

Comment:
@bkrausz tests pass, but an array of dictionaries has a pretty odd (ambiguous) encoding. `params` here is:

`"amount=100&source=btcrcv_test_receiver&currency=usd&level3[][red]=firstred&level3[][one]=fish&level3[][red]=another"`

But Stripe's API is happy with accepting this type of formatting (although different keys/values are needed for level3).

Do url encoders/decoders just hope that if there is an array of dictionaries, each dictionary has the same set of keys?